osu.Game/Rulesets/UI/HitObjectContainer.cs

```diff
@@ -147,6 +157,16 @@ public virtual void Add(DrawableHitObject hitObject)
             hitObject.OnRevertResult += onRevertResult;
 
             AddInternal(hitObject);
+
+            onDrawableHitObjectAddedRecursive(hitObject);
+        }
+
+        private void onDrawableHitObjectAddedRecursive(DrawableHitObject hitObject)
+        {
+            DrawableHitObjectAdded?.Invoke(hitObject);
+
+            foreach (var nested in hitObject.NestedHitObjects)
```

Comment:
This is weird, because it's not going to trigger on new nested DHOs, e.g. after sliders paths are changed to add more ticks. I expect this to be more important in catch than osu! with the changes made in this PR, so it deserves attention.

I think the source of even here should be from inside the DHO itself.